Fix functions

LiesaSalzer opened the pull request:
Hi @tnaake 
Thanks for your input! I implemented your suggestions in this new version. 
Now I think things should be fine.

LiesaSalzer committed:
- Added dependencies on DESCRIPTION: ggplot2 and magrittr (needed for mz_summary)
- Added mz_summary to general.R
- Show mz_summary in vignette
- Added example annotation file (x_annotation)
- Show in vignette how to add annotations to AdjacencyMatrix using rowData

LiesaSalzer committed:
Change dplyr version

LiesaSalzer committed:
test function

LiesaSalzer committed:
test functions

LiesaSalzer committed:
test functions

LiesaSalzer committed:
test functions

LiesaSalzer committed:
test functions

LiesaSalzer committed:
test functions

LiesaSalzer committed:
Test functions

LiesaSalzer committed:
Add descriptions and examples to the vignette

LiesaSalzer committed:
- mz_summary: added functionality to distinguish between combine and structural
- write unit tests test_mz_summary.R

LiesaSalzer committed:
- edit + split mz_summary
- smaller changes

LiesaSalzer committed:
update namespace and fix mz_summary

tnaake commented on R/mz_summary.R:
```diff
@@ -0,0 +1,185 @@
+#' `structural`or `combine`
+#' 
+#' @param filter
+#' `number`/`FALSE`, leave empty or set to `FALSE` if unfiltered data are 
```
please change `number` to `numeric`. Would it not be more clear to have instead of `FALSE` `0` as default? Could you please add that filter works on `counts` ?

tnaake commented on R/mz_summary.R:
```diff
@@ -0,0 +1,185 @@
+#'                                 mass = as.numeric(transformation[, 3]))
+#' am_struct <- structural(x_test, transformation, ppm = 5, directed = TRUE)
+#' mz_summary(am_struct)
+#'
```
Could you please add an example here as well on how to use `filter`?

tnaake commented on R/mz_summary.R:
```diff
@@ -0,0 +1,185 @@
+  if (!1 %in% x_df$binary) 
+    stop("'x' does not contain any mass-differences")
+  
+  if (!is.numeric(filter) & !FALSE %in% filter) 
```
When we enter a negative value for `filter` there is no warning/error. Is it not better to only allow for `0`/positive` integers here and return an error otherwise?

tnaake commented on R/mz_summary.R:
```diff
@@ -0,0 +1,185 @@
+#'
+#'@export
+#'
+#'@importFrom dplyr %>%
```
could you please put %>% in back-quotes: \`%>%` (cf the double entry of %>% in NAMESPACE)

tnaake commented on R/mz_summary.R:
```diff
@@ -0,0 +1,185 @@
+  x_df <- as.data.frame(x)
+  
+  if (!"combine" %in% type(x) & !"structural" %in% type(x) ) 
+    stop("'x' does not contain a mass-difference adjacency matrix")
```
Would it be more clear to have a message that `x` is not of type `combine` or `structural`?

tnaake commented on R/mz_summary.R:
```diff
@@ -0,0 +1,185 @@
+#'@export
+#'
+#'@importFrom dplyr %>%
+mz_summary <- function(x, filter = FALSE, ...){
```
Could you please use `am` instead of `x`? In this case we have `am` for all `AdjacencyMatrix` objects and `x` is refering to matrices.

tnaake replied:
where do you use `...`? this is not used in `mz_summary`, please remove.

tnaake commented on R/mz_summary.R:
```diff
@@ -0,0 +1,185 @@
+    stop("'x' does not contain a mass-difference adjacency matrix")
+  
+  if (!1 %in% x_df$binary) 
+    stop("'x' does not contain any mass-differences")
```
What happens if (for some reason) the `binary` assay in a `AdjacencyMatrix` of type `combine` is set to 0?
e.g.
    
    am_struct <- structural(x_test, transformation, ppm = 5, directed = TRUE)
    x <- x_test[,3:ncol(x_test)]
    x <- as.matrix(x)
    am_statistical <- statistical(x = x, model = c("pearson", "spearman"))
    am_threshold <- threshold(am_statistical, type = "threshold", args = list(filter = "pearson_coef > 0.1"))
    am_combine <- MetNet::combine(am_struct, am_threshold)
    assay(am_combine, "binary")[assay(am_combine, "binary") == 1] <- 0
    mz_summary(am_combine)

    Error in mz_summary(am_combine) : 
              'x' does not contain any mass-differences
              

I suggest to shift this check within the `if`/`else` and check for this case in the unit tests

LiesaSalzer replied:
I am not sure if I got your comment right.. So I was thinking of a function that checks if `combine` or `structural` contain any mass differences (that means any `!1 %in% x_df$binary`) at all. If not you will get this error message. Does it make sense?

tnaake commented on R/mz_summary.R:
```diff
@@ -0,0 +1,185 @@
+  sum_transform <- sum_transform %>% 
+    dplyr::select(transformation, mass_difference, counts)
+  
+  if (filter == FALSE | filter <= 1) {
```
I guess if you use only `0`/positive integer for `filter` then the `if/else` statement is not needed and code simplifies to 

    sum_f <- filter(sum_transform,sum_transform$counts >= filter)
    return(sum_f)

tnaake commented on R/mz_summary.R:
```diff
@@ -0,0 +1,185 @@
+#' @name mz_summary
+#'
+#' @aliases mz_summary
```
this tag is not needed

tnaake commented on R/mz_summary.R:
```diff
@@ -0,0 +1,185 @@
+
+#' @name mz_vis
+#'
+#' @aliases mz_vis
```
this tag is not needed

tnaake commented on R/mz_summary.R:
```diff
@@ -0,0 +1,185 @@
+      ggplot2::labs(title = "Numbers of determined mass differences")
+    
+    plot(plot_list)
+  
```
is there any specific reason why to assign the `ggplot2::...` expression to `plot_list` and use then `plot(plot_list)`? Why not having only the `ggplot2::...` expression here?

tnaake commented on R/x_annotation-data.R:
```diff
@@ -0,0 +1,44 @@
+#' @docType data
+#' @return \code{matrix}
+#' @format \code{matrix}
+#' @source
```
please add spaces between after each tag to improve readibility

tnaake commented on tests/testthat/test_mz_summary.R:
```diff
@@ -0,0 +1,72 @@
+
+struct_adj_neg <- structural(mat_test,
+                             transformation = transformations_neg, ppm = 5, directed = FALSE)
+
```
could you also add a test where `directed = FALSE` and check if `mz_summary` returns "only half the numbers"?

tnaake commented on tests/testthat/test_mz_summary.R:
```diff
@@ -0,0 +1,72 @@
+               c(4,4))
+  expect_true(is.data.frame(summary_cons_adj))
+})
+## END unit test mz_summary ##
```
could you also add tests please for `mz_vis`? e.g. check that it works with the correct input/ fails with the wrong input, returns a `ggplot` object, etc.

tnaake commented on vignettes/MetNet.Rmd:
````diff
@@ -312,6 +312,40 @@ struct_adj <- rtCorrection(am = struct_adj, x = x_test,
+
+```{r,eval=TRUE,echo=TRUE}
+mz_summary(struct_adj)
+```
````
please add an example on how to use `filter`

tnaake reviewed (changes requested):
Hi @LiesaSalzer 

many thanks! Looks great. There are some small things left. Please address them within this pull request #9 and once this is done, I will merge.

Great work and many thanks!

LiesaSalzer committed:
- mz_summary: changed object 'x' to 'am', set default of filter to '0', and rework error messages
- extend tests for mz_summary
- added tests for mz_vis
- added examples to run 'filter' in mz_summary(), and show mz_vis in vignette

LiesaSalzer committed:
-minor changes on test_mz_summary

LiesaSalzer commented:
Hi @tnaake I just made a commit trying to fix all remaining issues.

tnaake commented on R/mz_summary.R:
```diff
@@ -0,0 +1,171 @@
+#' which has been summarized by `mz_summary`. 
+#' 
+#' @param x
+#' `list`, previously generated by `mz_summary`. Needs to contain
```
one small note here that I just noticed. `mz_summary` returns a `data.frame`. I guess it would also be better to rename `list` to `data.frame`, use `df` or something similar instead of `x`  and check within the function that `df` is a `data.frame`:

    if (!is.data.frame(df)) stop(...)

LiesaSalzer replied:
Woops! You are absolutely right. I think it was a list some time ago. I will change that.

tnaake commented on R/mz_summary.R:
```diff
@@ -0,0 +1,171 @@
+  am_df <- as.data.frame(am)
+  
+  if (!"combine" %in% type(am) & !"structural" %in% type(am) ) 
+    stop("'am' is not a type 'structural' or 'combine'")
```
I hope this is not too pettily but could you please rename to 
   "'am' is not of type 'structural' or 'combine'"

tnaake commented on R/mz_summary.R:
```diff
@@ -0,0 +1,171 @@
+  if (!"combine" %in% type(am) & !"structural" %in% type(am) ) 
+    stop("'am' is not a type 'structural' or 'combine'")
+  
+  if (!1 %in% am_df$binary) 
```
> I am not sure if I got your comment right.. So I was thinking of a function that checks if combine or structural contain any mass differences (that means any !1 %in% x_df$binary) at all. If not you will get this error message. Does it make sense?

Yes, I like this in general to test for that. But if the user modifies the object (as I did in the example, then the function will stop even though there are links in `am_df$combine_binary`. I would suggest to test separately for `am_df$binary` and `am_df$combine_binary` within each `if`/`else` statement

    if ("combine" %in% type(am)) {
        if (!1 %in% am_df$binary) 
            stop(...)
        ....
    } 
   
    else {
         if (!1 %in% am_df$combine_binary)
             stop(...)
        ....
    }

LiesaSalzer replied:
Now I see. But for type `"combine"` I am using only the mass differences from `am_df$combine_binary.` Otherwise it would be the same as from type `"structural"` right?

LiesaSalzer replied:
Therefore I would like to test: 
```

  if ("combine" %in% type(am)) {
    if (!1 %in% am_df$combine_binary) 
      stop("'am' does not contain any mass-differences")
  } 
  
  else {
    if (!1 %in% am_df$binary)
      stop("'am' does not contain any mass-differences")
  }
```

tnaake replied:
Oh yes, I have flipped the if statements. Yours is correct - sorry

tnaake reviewed (changes requested):
One more round to go - almost there 👍

LiesaSalzer committed:
- mz_summary: Change filter default from FALSE to 0
- Other minor changes on mz_summary, mz_vis and test_mz_summary.R
